### PR TITLE
ability to build macos

### PR DESCRIPTION
**Pull request by casperstorm**
This PR makes it possible publish an release to drafts.
Currently only supporting macOS.

```
sh scripts/build-macos.sh dmg
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,65 @@
+TARGET = halloy
```
Comment (tarkah) [inline Makefile:1]: Do we plan on having linux commands in the Makefile ?

Comment (casperstorm) [reply Makefile:1]: I think it make sense to have it all there - think Mac should be the most complicated because .App and .Dmg?

Comment (tarkah) [reply Makefile:1]: Do you plan on using the makefile for anything outside of CI? I would never use it normally for just building / testing the app. Maybe we just make this a script `build-macos` and throw it in `scripts` folder?

Comment (tarkah) [reply Makefile:1]: Just feel like if ppl see `Makefile` in the root of our repo, they'll expect it to work on Linux too and I don't really want to use `Makefile` for that

**Comment by casperstorm** [comment]
@tarkah take a look at the updated script in `assets`.

**Comment by tarkah** [comment]
> @tarkah take a look at the updated script in `assets`.

Looks great!

**Review by tarkah — APPROVED** [review]

**Pull request merged**